app/models/concerns/vacancy_application_detail_validations.rb

```diff
@@ -12,17 +12,13 @@ module VacancyApplicationDetailValidations
   end
 
   def validity_of_publish_on
-    errors.add(:publish_on, publish_on_before_today_error) if publish_on_in_past? && publish_on_check?
+    errors.add(:publish_on, publish_on_before_today_error) if publish_on_in_past?
```

Comment:
This condition is used to prevent displaying errors messages for published vacancies with a past publication date, so now the message is incorrectly rendered on the edit form.

<img width="1182" alt="screen shot 2018-10-10 at 14 17 51" src="https://user-images.githubusercontent.com/159200/46738985-6e73a200-cc97-11e8-8064-1ac15b7c7f4d.png">